### PR TITLE
Skip current-deploy parameters

### PR DESCRIPTION
**Pull request by ulziibay**
This make sure parameters ending with `current-deploy` is not surfaced from `ark list`

### CODE REVIEW

File: store/parameter_store.go [hunk 1]
```diff
@@ -292,7 +295,11 @@ func (s *ParameterStore) List(env Environment, service string) ([]SecretIdentifi
 				return []SecretIdentifier{}, err
 			}
 			for _, result := range resp.Parameters {
-				ident, _ := getSecretIDFromParamName(*result.Name)
+				ident, err := getSecretIDFromParamName(*result.Name)
+				if err != nil {
+					// if any secrets cannot be converted, skip
+					continue
+				}
```
Comment (mcab) [inline store/parameter_store.go:302]: I'm admittedly not a huge fan of this. I know that we were ignoring these before, but now that we have a different case for the error (invalid environment versus ignore current deploy), there's room to make it explicit that we wish to skip certain types.

If we create two new errors (maybe `InvalidEnvironmentError` and `CurrentDeployError`), we can explicitly skip here on the latter. `InvalidEnvironmentError` would be on the interface, while `CurrentDeployError` would be only for Parameter Store. What do you think?

Comment (ulziibay) [reply store/parameter_store.go:302]: Will do!

Comment (ulziibay) [reply store/parameter_store.go]: Addressed. Pleasse take a look agian!

File: store/interface.go [hunk 2]
```diff
@@ -148,6 +148,25 @@ func (e *InvalidIdentifierError) Error() string {
...
+// CurrentDeployError occurs when a parameter name has suffix current-deploy.
+// Such parameters are private to catapult service and should not be surfaced via interface
+type CurrentDeployError struct {
+	Identifier string
+}
+
+func (e *CurrentDeployError) Error() string {
+	return fmt.Sprintf("current-deploy parameter should not be surfaced for parameter %s", e.Identifier)
+}
+
```
Comment (mcab) [inline store/interface.go:169]: Not sure if we need to have this within the general case. `current-deploy` is specific to the ParameterStore implementation. Up to you to keep it here or move it.

File: store/parameter_store.go [hunk 3]
```diff
@@ -292,7 +295,11 @@ func (s *ParameterStore) List(env Environment, service string) ([]SecretIdentifi
 				return []SecretIdentifier{}, err
 			}
 			for _, result := range resp.Parameters {
-				ident, _ := getSecretIDFromParamName(*result.Name)
+				ident, err := getSecretIDFromParamName(*result.Name)
+				if _, ok := err.(*CurrentDeployError); ok {
+					// if any secrets cannot be converted due to CurrentDeployError, skip
+					continue
+				}
```
Comment (mcab) [inline store/parameter_store.go:302]: nit: This comment relates to the processing a list of parameters, which means this error should be per parameter it processes. We can be more explicit about the intent.

```suggestion
				// secrets that fail with CurrentDeployError are intended to be read by machines, and not returned for human consumption.
				if _, ok := err.(*CurrentDeployError); ok {
					continue
				}
```

**Review by mcab — APPROVED** [review]
Small nits, but LGTM.

File: store/interface.go (hunk 2 above)
Comment (ulziibay) [reply store/interface.go:169]: Ok. wil move

**Pull request merged**